concurrent/src/main/scala/scalaz/concurrent/Task.scala

```diff
@@ -52,13 +53,23 @@ class Task[+A](val get: Future[Throwable \/ A]) {
    * This would typically be used to release any resources acquired by this
    * `Task`.
    */
-  def onFinish(f: Option[Throwable] => Task[Unit]): Task[A] =
-    new Task(get flatMap {
-      case -\/(e) =>
-        Task.Try(f(Some(e))).fold(e2 => Future.now(-\/(e2)), _.get *> Future.now(-\/(e)))
-      case r =>
-        Task.Try(f(None)).fold(e => Future.now(-\/(e)), _.get *> Future.now(r))
-    })
+   def onFinish(f: Option[Throwable] => Task[Unit]): Task[A] = {
+    import scalaz.std.function._
+    import scalaz.concurrent.Future._
+    val ttf = f.map(_.get.map(_ => ()))
```

Comment:
```Task { 4 }.onFinish(i => Task.delay { sys.error("blah") } )```

In the current version it has the same behavior, so while we can change it going forward, I'm not sure we should...